library/lua/dfhack.lua

```diff
@@ -636,33 +688,54 @@ function dfhack.gui.getViewscreenByType(scr_type, n)
     end
 end
 
+---@nodiscard
+---@return world_site|nil
 function dfhack.world.getCurrentSite()
     return df.world_site.find(df.global.plotinfo.site_id)
 end
 
+---@nodiscard
+---@param which string
+---@param key string
+---@param default? any
+---@return any
 local function persistent_getData(which, key, default)
     local serialized = dfhack.persistent['get'..which..'DataString'](key)
     if not serialized then return default end
     return require('json').decode(serialized) or default
 end
 
+---@param which string
+---@param key string
+---@param data any
 function persistent_saveData(which, key, data)
```

Comment:
`persistent_saveData` should be `local` like `persistent_getData` -- could you make that fix?